src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -306,53 +211,75 @@ private DynamicJsonValue GetSubscriptionConnectionOptions(SubscriptionConnection
             };
         }
 
-        public IEnumerable<DynamicJsonValue> GetAllRunningSubscriptions(DocumentsOperationContext context, bool history, int start, int take)
-        {
-            var transactionPersistentContext = new TransactionPersistentContext();
-            using (var tx = _environment.ReadTransaction(transactionPersistentContext))
+        public IEnumerable<DynamicJsonValue> GetAllRunningSubscriptions(TransactionOperationContext context, bool history, int start, int take)
+        {            
+            var databaseRecord = _serverStore.Cluster.Read(context, _db.Name);
+                
+            foreach (var kvp in _subscriptionStates)
             {
-                foreach (var kvp in _subscriptionStates)
-                {
-                    var subscriptionState = kvp.Value;
-                    var subscriptionId = kvp.Key;
+                var subscriptionState = kvp.Value;
+                var subscriptionId = kvp.Key;
 
-                    if (subscriptionState?.Connection == null)
-                        continue;
+                if (subscriptionState?.Connection == null)
+                    continue;
 
-                    if (start > 0)
-                    {
-                        start--;
-                        continue;
-                    }
+                if (start > 0)
+                {
+                    start--;
+                    continue;
+                }
 
-                    if (take-- <= 0)
-                        yield break;
+                if (take-- <= 0)
+                    yield break;
+                
+                BlittableJsonReaderObject currentSubscriptionReader = 
+                    GetSubscriptionFromDatabaseRecordBlittable(databaseRecord, subscriptionId);                
 
-                    yield return GetRunningSubscriptionInternal(context, history, subscriptionId, tx, subscriptionState);
-                }
+                Debug.Assert(currentSubscriptionReader != null);
+
+                yield return GetRunningSubscriptionInternal(history, currentSubscriptionReader, subscriptionState);
             }
+
         }
 
-        public unsafe DynamicJsonValue GetSubscription(DocumentsOperationContext context, long id, bool history)
+        private static BlittableJsonReaderObject GetSubscriptionFromDatabaseRecordBlittable(BlittableJsonReaderObject databaseRecord, long subscriptionId)
```

Comment:
Huh? What are you doing here? Looks like a binary search, but I can't figure out why you'll want / need that.
Leaving aside the fact that the `Subscriptions` is not sorted, you can just access the value directly.
Regardless, just get the `DatabaseRecord` and then use the `TryGetValue` on it.